### PR TITLE
Parse docx in --doc mode

### PR DESCRIPTION
**Pull request by babenek**
## Description

Please include a summary of the change and which is fixed.

based on #441

- Add scanner to parse docx files in text representation
- separated "documents" file types to obtain the results from file extractor
- refactored test samples for small size to fit in fuzzing process

## How has this been tested?

Please describe the tests that you ran to verify your changes.

- [x] UnitTest
- [x] Benchmark - no impact on benchmark

### CODE REVIEW

File: credsweeper/deep_scanner/docx_scanner.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+                for line in paragraph.text.splitlines():
+                    if line:
+                        docx_lines.append(line)
+
+            string_data_provider = StringContentProvider(lines=docx_lines,
+                                                         file_path=data_provider.file_path,
+                                                         file_type=data_provider.file_type,
+                                                         info=f"{data_provider.info}|DOCX")
+            pdf_candidates = self.scanner.scan(string_data_provider)
+            candidates.extend(pdf_candidates)
```
Comment (babenek) [inline credsweeper/deep_scanner/docx_scanner.py:41]:
```suggestion
            candidates = self.scanner.scan(string_data_provider)
            candidates.extend(candidates)
```

File: credsweeper/deep_scanner/docx_scanner.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+class DocxScanner(AbstractScanner, ABC):
+    """Implements pdf scanning"""
+
+    def data_scan(
+            self,  #
+            data_provider: DataContentProvider,  #
+            depth: int,  #
+            recursive_limit_size: int) -> List[Candidate]:
+        """Tries to scan PDF elements recursively and the whole text on page as strings"""
```
Comment (babenek) [inline credsweeper/deep_scanner/docx_scanner.py:24]:
```suggestion
        """Tries to scan DOCX text with splitting by lines"""
```

File: credsweeper/deep_scanner/docx_scanner.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+from credsweeper.credentials import Candidate
+from credsweeper.deep_scanner.abstract_scanner import AbstractScanner
+from credsweeper.file_handler.data_content_provider import DataContentProvider
+from credsweeper.file_handler.string_content_provider import StringContentProvider
+
+logger = logging.getLogger(__name__)
+
+
+class DocxScanner(AbstractScanner, ABC):
+    """Implements pdf scanning"""
```
Comment (babenek) [inline credsweeper/deep_scanner/docx_scanner.py:17]:
```suggestion
    """Implements docx scanning"""
```

File: credsweeper/secret/config.json [hunk 4]
```diff
@@ -71,6 +73,7 @@
             "/__pycache__/",
             "/node_modules/",
             "/target/",
+            "/.venv/",
```
[the remark this answers is not in the document]
Comment (babenek) [reply credsweeper/secret/config.json:76]: In my habit, all system folders i named with dot - it helps avoid it from git with simple rule - all hidden files.

[the remark this answers is not in the document]
Comment (csh519) [reply credsweeper/secret/config.json:76]: Okay. I should to consider to change my folder too.

**Review by csh519 — APPROVED** [review]
docx and pdf files scanning on `--doc` option has been added.

LGTM :+1:

**Review by Yullia — APPROVED** [review]

**Review by csh519 — APPROVED** [review]
Approve again.

**Review by kmnls — APPROVED** [review]
Agree

**Pull request merged**